Does anyone more familiar with typing know why mypy doesn't know about distributions?
```
scipy/stats/tests/test_entropy.py:274: error: Module has no attribute "norm"  [attr-defined]
scipy/stats/tests/test_entropy.py:275: error: Module has no attribute "expon"  [attr-defined]
```
We have a few `type: ignore[attr-defined]` for this reason, but I wonder if there's a way to fix it permanently.

--- a/scipy/stats/_entropy.py
+++ b/scipy/stats/_entropy.py
@@ -387,8 +387,8 @@ def _van_es_entropy(X, m, *, xp):
     n = X.shape[-1]
     difference = X[..., m:] - X[..., :-m]
     term1 = 1/(n-m) * xp.sum(xp.log((n+1)/m * difference), axis=-1)
-    k = xp.arange(m, n+1.)
-    return term1 + xp.sum(1./k) + math.log(m) - math.log(n+1)
+    k = xp.arange(m, n+1, dtype=term1.dtype)
+    return term1 + xp.sum(1/k) + math.log(m) - math.log(n+1)
 
 
 def _ebrahimi_entropy(X, m, *, xp):
@@ -399,7 +399,7 @@ def _ebrahimi_entropy(X, m, *, xp):
 
     differences = X[..., 2 * m:] - X[..., : -2 * m:]
 
-    i = xp.arange(1., n+1, dtype=X.dtype)
+    i = xp.arange(1, n+1, dtype=X.dtype)
     ci = xp.ones_like(i)*2
     ci[i <= m] = 1 + (i[i <= m] - 1)/m
     ci[i >= n - m + 1] = 1 + (n - i[i >= n-m+1])/m

--- a/scipy/stats/_entropy.py
+++ b/scipy/stats/_entropy.py
@@ -320,7 +320,7 @@ def differential_entropy(
     xp = array_namespace(values)
     values = xp.asarray(values)
     values = xp_moveaxis_to_end(values, axis, xp=xp)
-    n = values.shape[-1]  # number of observations
+    n = values.shape[-1]  # type: ignore[union-attr]
 
     if window_length is None:
         window_length = math.floor(math.sqrt(n) + 0.5)

--- a/scipy/stats/tests/test_entropy.py
+++ b/scipy/stats/tests/test_entropy.py
@@ -276,7 +276,7 @@ class TestDifferentialEntropy:
 
     @pytest.mark.parametrize('method', ['vasicek', 'van es', 'ebrahimi', 'correa'])
     @pytest.mark.parametrize('dist', [stats.norm, stats.expon])
-    def test_norm_rmse_std(self, method, dist, xp):
+    def test_rmse_std(self, method, dist, xp):
         # test that RMSE and standard deviation of estimators matches values
         # given in differential_entropy reference [6]. Incidentally, also
         # tests vectorization.

--- a/scipy/stats/tests/test_entropy.py
+++ b/scipy/stats/tests/test_entropy.py
@@ -5,6 +5,7 @@ from pytest import raises as assert_raises
 import numpy as np
 
 from scipy import stats
+from scipy.stats import norm, expon  # type: ignore[attr-defined]
 from scipy.conftest import array_api_compatible
 from scipy._lib._array_api import (xp_assert_close, xp_assert_equal, xp_assert_less,
                                    is_jax, is_array_api_strict, array_namespace)
@@ -271,11 +272,11 @@ class TestDifferentialEntropy:
                             'ebrahimi': (0.151, 0.148)
                             }
 
-    rmse_std_cases = {stats.norm: norm_rmse_std_cases,
-                      stats.expon: expon_rmse_std_cases}
+    rmse_std_cases = {norm: norm_rmse_std_cases,
+                      expon: expon_rmse_std_cases}
 
     @pytest.mark.parametrize('method', ['vasicek', 'van es', 'ebrahimi', 'correa'])
-    @pytest.mark.parametrize('dist', [stats.norm, stats.expon])
+    @pytest.mark.parametrize('dist', [norm, expon])
     def test_rmse_std(self, method, dist, xp):
         # test that RMSE and standard deviation of estimators matches values
         # given in differential_entropy reference [6]. Incidentally, also

